main.go

```diff
@@ -83,3 +117,19 @@ func VerboseLog(msg string) {
 		fmt.Println("*", msg)
 	}
 }
+
+func WriteStringProperty(b *bytes.Buffer, name string, value string) {
+	if value != "" {
+		WriteProperty(b, name, value)
+	}
+}
+
+func WriteUInt8Property(b *bytes.Buffer, name string, value uint8) {
+	if value != 0 {
```

Comment:
`CorrelationId` is [a `string`](https://godoc.org/github.com/streadway/amqp#Delivery).

 `DeliveryMode` can only be `1` or `2`.  `Priority` can be 0-9, with 0 as the default.
Technically [these are AMQP headers](http://docs.oasis-open.org/amqp/core/v1.0/amqp-core-messaging-v1.0.html#type-header) -durable and priority respectively. Which brings me back to my original question:

> AMQP makes a distinction between header and properties, and think headers 
> should be added too, but should there be separate options these and should they be in separate > files?

> If they should be in the same file then maybe it makes sense to dump the output to JSON.

:question: